rocAL/include/readers/file_source_reader.h

```diff
@@ -83,30 +85,42 @@ class FileSourceReader : public Reader {
     unsigned _curr_file_idx;
     FILE *_current_fPtr;
     unsigned _current_file_size;
+    unsigned _shard_start_idx;
+    std::vector<unsigned> _shard_start_idx_vector, _shard_end_idx_vector;
     std::string _last_id;
-    std::string _last_file_name, _last_file_path;
+    std::string _last_file_name, _last_file_path, _absolute_file_path;
     size_t _shard_id = 0;
     size_t _shard_count = 1;  // equivalent of batch size
-    //!< _batch_count Defines the quantum count of the images to be read. It's usually equal to the user's batch size.
-    /// The loader will repeat images if necessary to be able to have images available in multiples of the load_batch_count,
-    /// for instance if there are 10 images in the dataset and _batch_count is 3, the loader repeats 2 images as if there are 12 images available.
-    size_t _batch_count = 1;
+    signed _shard_size = -1;
```

Comment:
we need to convert signed and unsigned to int32_t and uint32_t. For now please use that for all the new variables added. We cab create a new PR later for data_type conversion

Reply:
done